pykickstart/commands/displaymode.py

```diff
@@ -56,7 +57,54 @@ def parse(self, args):
         (_opts, extra) = self.op.parse_args(args=args, lineno=self.lineno)
 
         if len(extra) > 0:
-            raise KickstartParseError(formatErrorMsg(self.lineno, msg=_("Kickstart command %s does not take any arguments") % self.currentCmd))
+            msg = _("Kickstart command %s does not take any arguments") % self.currentCmd
+            raise KickstartParseError(formatErrorMsg(self.lineno, msg))
+
+        if self.currentCmd == "cmdline":
+            self.displayMode = DISPLAY_MODE_CMDLINE
+        elif self.currentCmd == "graphical":
+            self.displayMode = DISPLAY_MODE_GRAPHICAL
+        elif self.currentCmd == "text":
+            self.displayMode = DISPLAY_MODE_TEXT
+
+        return self
+
+class F26_DisplayMode(FC3_DisplayMode):
+    removedKeywords = KickstartCommand.removedKeywords
+    removedAttrs = KickstartCommand.removedAttrs
+
+    def __init__(self, writePriority=0, *args, **kwargs):
+        FC3_DisplayMode.__init__(self, writePriority, args, kwargs)
+        self.op = self._getParser()
+        self.displayMode = kwargs.get("displayMode", None)
+        self.nonInteractive = kwargs.get("nonInteractive", False)
+
+    def __str__(self):
+        retval = super(F26_DisplayMode, self).__str__()
+
+        if self.nonInteractive:
+            retval = retval[:-1]
+            retval += " --non-interactive\n"
+
+        return retval
+
+    def _getParser(self):
+        op = FC3_DisplayMode._getParser(self)
+        op.add_option("--non-interactive", action="store_true", default=False,
+                      dest="nonInteractive")
+        return op
+
+    def parse(self, args):
+        (opts, extra) = self.op.parse_args(args=args, lineno=self.lineno)
+        self.set_to_self(self.op, opts)
+
+        if len(extra) > 0:
+            msg = _("Kickstart command %s does not take any arguments") % self.currentCmd
+            raise KickstartParseError(formatErrorMsg(self.lineno, msg))
+
+        if self.currentCmd == "cmdline" and self.nonInteractive:
+            msg = _("Kickstart command cmdline does not support --non-interactive parameter")
+            raise KickstartParseError(formatErrorMsg(self.lineno, msg))
```

Comment:
I believe this should call the parent class' ``parse()`` method and just perform the extra checks.